src/txrebroadcast.h

```diff
@@ -8,15 +8,53 @@
 #include <policy/feerate.h>
 #include <tuple>
 #include <txmempool.h>
+#include <util/hasher.h>
+#include <util/time.h>
 #include <validation.h>
 
+#include <boost/multi_index/hashed_index.hpp>
+#include <boost/multi_index/member.hpp>
+#include <boost/multi_index/ordered_index.hpp>
+#include <boost/multi_index_container.hpp>
+
 struct TxIds {
     TxIds(uint256 txid, uint256 wtxid) : m_txid(txid), m_wtxid(wtxid) {}
 
     const uint256 m_txid;
     const uint256 m_wtxid;
 };
 
+struct RebroadcastEntry {
+    RebroadcastEntry(std::chrono::microseconds now_time, uint256 wtxid)
+        : m_last_attempt(now_time),
+          m_wtxid(wtxid),
+          m_count(1) {}
+
+    std::chrono::microseconds m_last_attempt;
+    const uint256 m_wtxid;
+    int m_count;
+};
+
+/** Used for multi_index tag  */
+struct index_by_last_attempt {};
+
+using indexed_rebroadcast_set = boost::multi_index_container<
```

Comment:
The mempool is persisted to disk, would it make sense to do the same with the rebroadcast tracker? With the index designed to keep a memory of months, I'd guess that several nodes would undergo several restarts during such a long period and, currently, lose all memory about past rebroadcasts.

Reply:
seems reasonable, but this PR is getting pretty huge so I'd rather keep this for a follow-up. I've noted it down here https://github.com/bitcoin/bitcoin/pull/21061#issuecomment-771426879 so I can keep track of it, and am going to resolve this conversation.